src/parser.ts

```diff
@@ -109,11 +104,16 @@ export class Parser extends DiagnosticEmitter {
   currentSource: Source;
   /** Dependency map **/
   dependees: Map<string, Source> = new Map();
+  /** An array of parsed sources. */
+  sources: Source[];
 
   /** Constructs a new parser. */
-  constructor(program: Program) {
-    super(program.diagnostics);
-    this.program = program;
+  constructor(
+    diagnostics: DiagnosticMessage[] | null = null,
+    sources: Source[] | null = null
+  ) {
+    super(diagnostics);
+    this.sources = sources == null ? new Array<Source>() : sources;
```

Comment:
```suggestion
    this.sources = sources ? sources : new Array<Source>();
```

Mostly curious since you might have chosen to write it this way due to a null checking issue. Is there one?

Reply:
I was following the code style in the parent class. I would have coded it `sources || new Array<Source>()`.